homeassistant/components/subaru/sensor.py

```diff
@@ -146,140 +161,119 @@ async def async_setup_entry(
 
     @property
     def native_value(self):
         """Return the state of the sensor."""
-        self.current_value = self.get_current_value()
+        current_value = self.get_current_value()
+        unit = self.entity_description.native_unit_of_measurement
+        unit_system = self.hass.config.units
 
-        if self.current_value is None:
+        if current_value is None:
             return None
 
-        if self.api_unit in TEMPERATURE_UNITS:
-            return round(
-                self.hass.config.units.temperature(self.current_value, self.api_unit), 1
-            )
+        if unit in LENGTH_UNITS:
+            return round(unit_system.length(current_value, unit), 1)
 
-        if self.api_unit in LENGTH_UNITS:
+        if unit in PRESSURE_UNITS and unit_system == IMPERIAL_SYSTEM:
             return round(
-                self.hass.config.units.length(self.current_value, self.api_unit), 1
-            )
-
-        if (
-            self.api_unit in PRESSURE_UNITS
-            and self.hass.config.units == IMPERIAL_SYSTEM
-        ):
-            return round(
-                self.hass.config.units.pressure(self.current_value, self.api_unit),
+                unit_system.pressure(current_value, unit),
                 1,
             )
 
-        if (
-            self.api_unit in FUEL_CONSUMPTION_UNITS
-            and self.hass.config.units == IMPERIAL_SYSTEM
-        ):
-            return round((100.0 * L_PER_GAL) / (KM_PER_MI * self.current_value), 1)
+        if unit in FUEL_CONSUMPTION_UNITS and unit_system == IMPERIAL_SYSTEM:
+            return round((100.0 * L_PER_GAL) / (KM_PER_MI * current_value), 1)
 
-        return self.current_value
+        return current_value
 
     @property
     def native_unit_of_measurement(self):
         """Return the unit_of_measurement of the device."""
-        if self.api_unit in TEMPERATURE_UNITS:
-            return self.hass.config.units.temperature_unit
+        unit = self.entity_description.native_unit_of_measurement
 
-        if self.api_unit in LENGTH_UNITS:
+        if unit in LENGTH_UNITS:
             return self.hass.config.units.length_unit
 
-        if self.api_unit in PRESSURE_UNITS:
+        if unit in PRESSURE_UNITS:
             if self.hass.config.units == IMPERIAL_SYSTEM:
                 return self.hass.config.units.pressure_unit
-            return PRESSURE_HPA
 
-        if self.api_unit in FUEL_CONSUMPTION_UNITS:
+        if unit in FUEL_CONSUMPTION_UNITS:
             if self.hass.config.units == IMPERIAL_SYSTEM:
                 return FUEL_CONSUMPTION_MPG
-            return FUEL_CONSUMPTION_L_PER_100KM
 
-        return self.api_unit
+        return unit
 
     @property
     def available(self):
         """Return if entity is available."""
         last_update_success = super().available
         if last_update_success and self.vin not in self.coordinator.data:
             return False
+        if self.state is None:
+            return False
```

Comment:
We translate state `None` to unknown state. Unavailable and unknown state means different things. We should not set state to unavailable if state is unknown. Unavailable state means we know that the device or service is offline. Unknown state means we don't know yet what the state is. We need to be able to keep these cases distinguishable.

If the 3rd party API returns a `None` as value *and* it means the device or service is unavailable we need to determine that separately.
```suggestion
```